### PR TITLE
update screenshot on the slice page when coming back to tab

### PR DESCRIPTION
**Pull request by mathieud**
When coming back to the tab a request to /state is triggered. This changes propagates changes to the screenshot to the `selectedSlice` reducer if a match is found, updating the screenshot a user sees.

## Types of changes

- [ x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

The `selectedSliceReducer` now listens to the `STATE/REFRESH.RESPONSE` action, and update the screenshot data **only** on the selected slice if found in the response.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ x] All new and existing tests are passing.

### CODE REVIEW

**Review by danigo99 — APPROVED** [review]

File: packages/slice-machine/tests/src/modules/selectedSlice/reducer.test.ts
Comment (MarcMcIntosh) [inline packages/slice-machine/tests/src/modules/selectedSlice/reducer.test.ts:178]: nice

**Review by MarcMcIntosh — APPROVED** [review]
I think we'll need to wait before merging :)

**Comment by mathieud** [comment]
> I think we'll need to wait before merging :)

@MarcMcIntosh it merges into Dan's dev branch (same feature/ticket), not a release version.  Do you still want to wait until we merge?

**Comment by MarcMcIntosh** [comment]
Cool, I thought it was going into the v0.4.3 branch

**Pull request merged**
